fuse_models/include/fuse_models/common/sensor_proc.h

```diff
@@ -706,6 +706,194 @@ inline bool processDifferentialPoseWithCovariance(
+ * @param[in] source - The name of the sensor or motion model that generated this constraint
+ * @param[in] device_id - The UUID of the machine
+ * @param[in] pose1 - The first (and temporally earlier) PoseWithCovarianceStamped message
+ * @param[in] pose2 - The first (and temporally later) PoseWithCovarianceStamped message
+ * @param[in] twist - The first (and temporally later) TwistWithCovarianceStamped message
+ * @param[in] minimum_pose_relative_covariance - The minimum pose relative covariance that is always added to the
+ *                                               resulting pose relative covariance
+ * @param[in] loss - The loss function for the 2D pose constraint generated
+ * @param[in] validate - Whether to validate the measurements or not. If the validation fails no constraint is added
+ * @param[out] transaction - The generated variables and constraints are added to this transaction
+ * @return true if any constraints were added, false otherwise
+ */
+inline bool processDifferentialPoseWithTwistCovariance(
+  const std::string& source,
+  const fuse_core::UUID& device_id,
+  const geometry_msgs::PoseWithCovarianceStamped& pose1,
+  const geometry_msgs::PoseWithCovarianceStamped& pose2,
+  const geometry_msgs::TwistWithCovarianceStamped& twist,
+  const fuse_core::Matrix3d& minimum_pose_relative_covariance,
+  const fuse_core::Loss::SharedPtr& loss,
+  const std::vector<size_t>& position_indices,
+  const std::vector<size_t>& orientation_indices,
+  const bool validate,
+  fuse_core::Transaction& transaction)
+{
+  if (position_indices.empty() && orientation_indices.empty())
+  {
+    return false;
+  }
+
+  // Convert the poses into tf2_2d transforms
+  tf2_2d::Transform pose1_2d;
+  tf2::fromMsg(pose1.pose.pose, pose1_2d);
+
+  tf2_2d::Transform pose2_2d;
+  tf2::fromMsg(pose2.pose.pose, pose2_2d);
+
+  // Create the pose variables
+  auto position1 = fuse_variables::Position2DStamped::make_shared(pose1.header.stamp, device_id);
+  auto orientation1 =
+    fuse_variables::Orientation2DStamped::make_shared(pose1.header.stamp, device_id);
+  position1->x() = pose1_2d.x();
+  position1->y() = pose1_2d.y();
+  orientation1->yaw() = pose1_2d.yaw();
+
+  auto position2 = fuse_variables::Position2DStamped::make_shared(pose2.header.stamp, device_id);
+  auto orientation2 = fuse_variables::Orientation2DStamped::make_shared(pose2.header.stamp, device_id);
+  position2->x() = pose2_2d.x();
+  position2->y() = pose2_2d.y();
+  orientation2->yaw() = pose2_2d.yaw();
+
+  // Create the delta for the constraint
+  const double sy = ::sin(-pose1_2d.yaw());
+  const double cy = ::cos(-pose1_2d.yaw());
+  double x_diff = pose2_2d.x() - pose1_2d.x();
+  double y_diff = pose2_2d.y() - pose1_2d.y();
+  fuse_core::Vector3d pose_relative_mean;
+  pose_relative_mean <<
+    cy * x_diff - sy * y_diff,
+    sy * x_diff + cy * y_diff,
+    (pose2_2d.rotation() - pose1_2d.rotation()).getAngle();
+
+  // Create the covariance components for the constraint
```

Comment:
The code above is the same as in the beginning of https://github.com/clearpathrobotics/fuse/blob/cpr-devel/fuse_models/include/fuse_models/common/sensor_proc.h#L389